pulpcore/cli/common/generic.py

```diff
@@ -256,10 +258,35 @@ def _option_callback(
         if not value:
             return value
 
-        split_value = value.split(":", maxsplit=2)
-        while len(split_value) < 3:
-            split_value.insert(0, "")
-        plugin, resource_type, identifier = split_value
+        pulp_href: Optional[str] = None
+        entity: Optional[EntityDefinition] = None
+
+        if value.startswith("/"):
```

Comment:
I wonder if this shouldn't check for href_pattern instead and then just assume it's a name if it doesn't match:

```
$ pulp file remote create --name /test/ --url https://fixtures.pulpproject.org/file/
$ pulp file repository create --name /test/ --remote /test/
Error: '/test/' is not a valid href for remote.
```

Reply:
That wouldn't solve the problem where the name actually looks like a valid href.
But you can always use: `$ pulp file repository create --name /test/ --remote ::/test/`